modules/dnn/src/layers/permute_layer.cpp

```diff
@@ -382,14 +401,30 @@ class PermuteLayerImpl CV_FINAL : public PermuteLayer
         return Ptr<BackendNode>();
     }
 
-#ifdef HAVE_INF_ENGINE
     virtual Ptr<BackendNode> initInfEngine(const std::vector<Ptr<BackendWrapper> >&) CV_OVERRIDE
     {
+#ifdef HAVE_INF_ENGINE
+#if INF_ENGINE_VER_MAJOR_GE(INF_ENGINE_RELEASE_2018R5)
         InferenceEngine::Builder::PermuteLayer ieLayer(name);
         ieLayer.setOrder(_order);
         return Ptr<BackendNode>(new InfEngineBackendNode(ieLayer));
-    }
+#else
+        InferenceEngine::LayerParams lp;
+        lp.name = name;
+        lp.type = "Permute";
+        lp.precision = InferenceEngine::Precision::FP32;
+        std::shared_ptr<InferenceEngine::CNNLayer> ieLayer(new InferenceEngine::CNNLayer(lp));
+
+        CV_Assert(!_order.empty());
+        ieLayer->params["order"] = format("%zu", _order[0]);
+        for (int i = 1; i < _order.size(); ++i)
+            ieLayer->params["order"] += format(",%zu", _order[i]);
+
+        return Ptr<BackendNode>(new InfEngineBackendNode(ieLayer));
```

Comment:
Looks like these changes are unrelated to this patch.
@dkurt Please take a look.

Reply:
It's definitely not related to this PR. I think it made its way in silently through a rebase.